{"schemaVersion":{"major":0,"minor":3,"patch":0},"metadata":{"role":"article","modules":[{"name":"VigramSDK"}],"title":"SDK Errors and exceptions","images":[{"identifier":"logo.png","type":"icon"}],"roleHeading":"Article"},"identifier":{"url":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SDKErrorsAndExceptions","interfaceLanguage":"swift"},"primaryContentSections":[{"content":[{"text":"","level":2,"anchor":"","type":"heading"},{"type":"paragraph","inlineContent":[{"text":"This section provides a detailed description of errors that may occur while using the SDK.","type":"text"}]},{"items":[{"content":[{"type":"paragraph","inlineContent":[{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","type":"reference","isActive":true}]}]}],"type":"unorderedList"},{"type":"paragraph","inlineContent":[{"text":"These errors can be received when using ","type":"text"},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Authentication","isActive":true}]},{"type":"codeListing","syntax":"swift","code":["public enum AuthenticationError: Swift.Error {","","    \/\/ Error authentication - bundle is not exist","    case bundleNotExist(String)","","    \/\/ Error authentication - token is not exist","    case tokenNotExist(String)","","    \/\/ Error authentication - wrong combination ","    \/\/ bundle and token is incorrect)","    case wrongTokenAndBundle(String)","","    \/\/ Error authentication - wrong combination ","    \/\/ (bundle, token or platformId is incorrect)","    case wrongCombination(String)","","    \/\/ Authentication is failure - Other error received ","    \/\/ from server","    case failure(String)","}"]},{"type":"unorderedList","items":[{"content":[{"type":"paragraph","inlineContent":[{"isActive":true,"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError"}]}]}]},{"type":"paragraph","inlineContent":[{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError","isActive":true},{"text":" combines different failure events that can be emitted on sending data to a ","type":"text"},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","type":"reference","isActive":true}]},{"code":["public enum PeripheralError: Swift.Error {","","    \/\/ The peripheral does not have a characteristic to write to.","    \/\/ Note: This error might be thrown, when we did not yet gather ","    \/\/ enough characteristics of the peripheral.","    case characteristicNotFound","","    \/\/ A send request is aborted, since the peripheral has already ","    \/\/ been deallocated.","    \/\/ Since VigramSDK is waiting for all previous messages to be ","    \/\/ sent before attempting","    \/\/ to send a message, it might already be deallocated when trying ","    \/\/ to send a message.","    case alreadyDeallocated","","    \/\/ Laser measurements not available in PPK mode","    case laserNotAvailableInPPK","","    \/\/ BLE write timeout exceeded.","    case writeTimeOut","","    \/\/ The configuration of the peripheral failed unexpectedly.","    case configurationFailed","    ","    \/\/ This method is not supported in current software version ","    \/\/ of the RTK device","    case requestIsNotSupported(String)","    ","    \/\/ This method is not available in this case","    case accessDenied(String)","    ","    \/\/ The current RTK device is incompatible in this SDK version","    case deviceNumberIsIncompatible","    ","    \/\/ The current software of the connected RTK device is ","    \/\/ incompatible in this SDK version","    case deviceSoftwareIsIncompatible","    ","    \/\/ Is not possible to perform the request. Device is configuring","    \/\/ in this time.","    case deviceIsBusy(String)","","    \/\/ The current version of SDK is outdated","    case sdkIsOutdated","","    \/\/ The verification of SDK is failure","    case verificationFailure(String)","","    \/\/ The unknown error","    case unknownError(String)","}"],"syntax":"swift","type":"codeListing"},{"type":"unorderedList","items":[{"content":[{"type":"paragraph","inlineContent":[{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","isActive":true}]}]}]},{"type":"paragraph","inlineContent":[{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","isActive":true},{"type":"text","text":" lists different errors that can be propagated by "},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","type":"reference","isActive":true},{"text":" implementations.","type":"text"}]},{"type":"codeListing","syntax":"swift","code":["public enum BluetoothServiceError: Error {","","    \/\/ The device to connect to cannot be found, since it is","    \/\/ (possibly no longer) available.","    case notAvailable(UUID)","","    \/\/ Connecting to a device failed unexplicably.","    case notConnected(UUID)","}"]},{"items":[{"content":[{"type":"paragraph","inlineContent":[{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","isActive":true}]}]}],"type":"unorderedList"},{"type":"paragraph","inlineContent":[{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","isActive":true},{"type":"text","text":" contains custom failure events when connecting to Ntrip services."}]},{"type":"paragraph","inlineContent":[{"type":"strong","inlineContent":[{"type":"text","text":"Note:"}]},{"type":"text","text":" It is not ensured that calls to "},{"isActive":true,"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripService","type":"reference"},{"text":" can only result in ","type":"text"},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","isActive":true,"type":"reference"},{"type":"text","text":" events."}]},{"syntax":"swift","code":["public enum NtripServiceError: Swift.Error {","","    \/\/ Mountpoints could not be decoded from the received data.","    case mountPointDecodingFailed","","    \/\/ Creating the URL to connect to Ntrip endpoints did not succeed.","    case couldNotCreateURL(NtripConnectionInformation, ","                           mountPoint: String?)","","    \/\/ We could not create an Ntrip task due to unexplicable ","    \/\/ circumstances.","    \/\/ Note: This error should never be thrown.","    case couldNotCreateTask","","    \/\/ Adding authentication to a URLRequest did not succeed.","    case couldNotAddAuthentication(URLRequest, username: String, ","                                   password: String)","}"],"type":"codeListing"},{"type":"unorderedList","items":[{"content":[{"type":"paragraph","inlineContent":[{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError","type":"reference","isActive":true}]}]}]},{"type":"paragraph","inlineContent":[{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError","isActive":true},{"type":"text","text":" lists different errors that can be emitted by "},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask","isActive":true},{"type":"text","text":" implementations."}]},{"type":"codeListing","syntax":"swift","code":["public enum NtripTaskError: Swift.Error {","","    \/\/ Creating the Ntrip network request failed.","    case couldNotCreateRequest","","    \/\/ Failed due to no data","    case noData","","    \/\/ Ntrip connection information (domain, port, or cridential) ","    \/\/ is incorrect","    case incorrectNCI","","    \/\/ Other error, received from RTCM message","    case failed(String)","","    \/\/ HTTP responce error, received from server","    case httpResponceError(String)","","    \/\/ Server is not responce error","    case serverIsNotResponce","}"]},{"type":"unorderedList","items":[{"content":[{"inlineContent":[{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError","isActive":true,"type":"reference"}],"type":"paragraph"}]}]},{"inlineContent":[{"isActive":true,"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError"},{"text":" are thrown, when due to the filtering phase of single point recordings there is not enough valid data to create a singlePoint. To investigate possible issues with the measurement, have a look at the provided associated data.","type":"text"}],"type":"paragraph"},{"syntax":"swift","code":["public enum SinglePointRecordingError: Error {","","\/\/ The measurement failed due to too much filtered, non-valid data.","\/\/ Most of the measured data have low accuracy.","\/\/ e.g. One or more conditions are not met","\/\/ Hor.Accuracy < Configuration.maximumSinglePointHorizontalAccuracy,","\/\/ Vert.Accuracy < Configuration.maximumSinglePointVerticalAccuracy","\/\/ Altitude < Configuration.maximumSinglePointAltitudeDifference","\/\/ The returned RAW value of this error is the received measurements.","case badMeasurement([EnvironmentData])","","\/\/ The measurement cancelled by user.","case measurementCancelled","","\/\/ The measurement failure.","\/\/ In most cases this error is not returned. ","\/\/ If you get it, check if you passed the correct ","\/\/ CoordinateCorrection.Method to the SinglePointRecordingService ","\/\/ measurement procedure (record or startMeasurements).","case measurementFailure","}"],"type":"codeListing"}],"kind":"content"}],"kind":"article","sections":[],"hierarchy":{"paths":[["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK"]]},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/vigramsdk\/sdkerrorsandexceptions"]}],"references":{"doc://com.vigram.VigramSDK/documentation/VigramSDK/Peripheral":{"url":"\/documentation\/vigramsdk\/peripheral","title":"Peripheral","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"Peripheral"}],"role":"symbol","navigatorTitle":[{"text":"Peripheral","kind":"identifier"}],"abstract":[{"text":"Peripheral provides a write and receive interface to communicate with the viDoc device.","type":"text"}],"kind":"symbol"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripTaskError":{"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError","type":"topic","title":"NtripTaskError","abstract":[{"type":"text","text":"NtripTaskError lists different errors that can be emitted by "},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask","isActive":true,"type":"reference"},{"text":" implementations.","type":"text"}],"role":"symbol","navigatorTitle":[{"text":"NtripTaskError","kind":"identifier"}],"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"NtripTaskError","kind":"identifier"}],"url":"\/documentation\/vigramsdk\/ntriptaskerror"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothService":{"url":"\/documentation\/vigramsdk\/bluetoothservice","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"text":"BluetoothService","kind":"identifier"}],"title":"BluetoothService","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","navigatorTitle":[{"text":"BluetoothService","kind":"identifier"}],"role":"symbol","kind":"symbol","type":"topic","abstract":[{"type":"text","text":"A BluetoothService is used to handle bluetooth connections."}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripTask":{"url":"\/documentation\/vigramsdk\/ntriptask","kind":"symbol","title":"NtripTask","abstract":[{"text":"A NtripTask requests RTK data over a NTRIP connection.","type":"text"}],"role":"symbol","fragments":[{"kind":"keyword","text":"protocol"},{"text":" ","kind":"text"},{"text":"NtripTask","kind":"identifier"}],"navigatorTitle":[{"text":"NtripTask","kind":"identifier"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask","type":"topic"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothServiceError":{"title":"BluetoothServiceError","fragments":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"text":"BluetoothServiceError","kind":"identifier"}],"abstract":[{"text":"BluetoothServiceError lists different errors that can be propagated by ","type":"text"},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","isActive":true,"type":"reference"},{"type":"text","text":" implementations."}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","url":"\/documentation\/vigramsdk\/bluetoothserviceerror","type":"topic","navigatorTitle":[{"kind":"identifier","text":"BluetoothServiceError"}],"role":"symbol","kind":"symbol"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/SinglePointRecordingError":{"role":"symbol","url":"\/documentation\/vigramsdk\/singlepointrecordingerror","kind":"symbol","type":"topic","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError","abstract":[{"type":"text","text":"SinglePointRecordingError are thrown, when due to the filtering phase of single point recordings there is not enough valid data to create a singlePoint."}],"title":"SinglePointRecordingError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SinglePointRecordingError"}],"navigatorTitle":[{"text":"SinglePointRecordingError","kind":"identifier"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripServiceError":{"navigatorTitle":[{"text":"NtripServiceError","kind":"identifier"}],"kind":"symbol","abstract":[{"text":"NtripServiceError contains custom failure events when connecting to Ntrip services.","type":"text"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NtripServiceError"}],"title":"NtripServiceError","role":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/ntripserviceerror"},"doc://com.vigram.VigramSDK/documentation/VigramSDK":{"title":"VigramSDK","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK","abstract":[],"kind":"symbol","images":[{"identifier":"logo.png","type":"icon"}],"role":"collection","url":"\/documentation\/vigramsdk","type":"topic"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError":{"kind":"symbol","url":"\/documentation\/vigramsdk\/authenticationerror","role":"symbol","navigatorTitle":[{"text":"AuthenticationError","kind":"identifier"}],"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AuthenticationError","kind":"identifier"}],"type":"topic","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","title":"AuthenticationError","abstract":[{"type":"text","text":"AuthenticationError contains custom failure events when connecting to Authentication process."}]},"logo.png":{"variants":[{"url":"\/images\/com.vigram.VigramSDK\/logo.png","traits":["1x","light"]}],"identifier":"logo.png","alt":"Vigram logo.","type":"image"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripService":{"fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"NtripService","kind":"identifier"}],"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripService","role":"symbol","url":"\/documentation\/vigramsdk\/ntripservice","type":"topic","title":"NtripService","abstract":[{"text":"NtripService is used to search for NTRIP mountpoints for various NTRIP providers and create ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask"},{"text":" instances to access NTRIP data.","type":"text"}],"navigatorTitle":[{"text":"NtripService","kind":"identifier"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/Authentication":{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Authentication","kind":"symbol","abstract":[{"text":"First creates a ","type":"text"},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Authentication","isActive":true},{"type":"text","text":" to check token and use SDK functions."}],"title":"Authentication","url":"\/documentation\/vigramsdk\/authentication","navigatorTitle":[{"kind":"identifier","text":"Authentication"}],"role":"symbol","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"Authentication"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/PeripheralError":{"kind":"symbol","url":"\/documentation\/vigramsdk\/peripheralerror","role":"symbol","navigatorTitle":[{"kind":"identifier","text":"PeripheralError"}],"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"PeripheralError"}],"type":"topic","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError","title":"PeripheralError","abstract":[{"type":"text","text":"PeripheralError combines different failure events that can be emitted on sending data to a "},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","isActive":true,"type":"reference"},{"text":".","type":"text"}]}}}